### DISCUSSION TITLE
useFormContext and useFieldArray returns value.filter is not a function with controller

### DISCUSSION THREAD

**Discussion by jostal** [react-hook-form/react-hook-form#7989] [Q&A]
I have a component which searches for data based on user input. This input is rendered in a Controller component. When I comment out this controller, there is no longer an issue. Also, when I switch to useForm() instead of useFormContext() there is no issue, but I need to use useFormContext() for this component, so that is not a valid fix.
The error returned is 'value.filter is not a function'. Following the source, the error occurs in useFieldArray and it does appear to be the case that it is only an issue when using useFormContext and useFieldArray together with this controller. I'm not filtering anything from the fieldArray so I'm not sure what the issue is.
Ex/
```
cont {control} = useFormContext();
const fieldArray = useFieldArray({ control, name: 'namehere' })

return (
  <Controller 
    control={control}
    name={name}
    defaultValue=""
    render={({
      fieldState,
      field: { value, name, ref }
    }) => {
      return (
        <SearchBox 
          name={name}
          value={value}
          onChange={() => /*if product is selected assigns to fieldArray**/}
          ref={ref}
        />
      )
    }}
  />
)
```
Any idea why when using these together it would return value.filter is not a function?
To clarify, I use this component both for unpopulated and prepopulated forms (based on saved data).
Any help is appreciated, thanks in advance.

**Comment by jostal** [comment]
After some testing, it appears to be a problem with the defaultValue in the controller. When I remove it everything works as intended. Any idea why that is?

**Comment by bluebill1049** [comment]
you don't need `defaultValue` with `useFieldArray` with `Controller` or `register`, because data will be feed by the `useFieldArray`
